addons/account/models/chart_template.py

```diff
@@ -1,991 +1,680 @@
-            ("asset_fixed", "Fixed Assets"),
-            ("liability_payable", "Payable"),
-            ("liability_credit_card", "Credit Card"),
-            ("liability_current", "Current Liabilities"),
-            ("liability_non_current", "Non-current Liabilities"),
-            ("equity", "Equity"),
-            ("equity_unaffected", "Current Year Earnings"),
-            ("income", "Income"),
-            ("income_other", "Other Income"),
-            ("expense", "Expenses"),
-            ("expense_depreciation", "Depreciation"),
-            ("expense_direct_cost", "Cost of Revenue"),
-            ("off_balance", "Off-Balance Sheet"),
-        ],
-        string="Type",
-        help="These types are defined according to your country. The type contains more information "\
-        "about the account and its specificities."
-    )
-    reconcile = fields.Boolean(string='Allow Invoices & payments Matching', default=False,
-        help="Check this option if you want the user to reconcile entries in this account.")
-    note = fields.Text()
-    tax_ids = fields.Many2many('account.tax.template', 'account_account_template_tax_rel', 'account_id', 'tax_id', string='Default Taxes')
-    nocreate = fields.Boolean(string='Optional Create', default=False,
-        help="If checked, the new chart of accounts will not contain this by default.")
-    chart_template_id = fields.Many2one('account.chart.template', string='Chart Template',
-        help="This optional field allow you to link an account template to a specific chart template that may differ from the one its root parent belongs to. This allow you "
-            "to define chart templates that extend another and complete it with few new accounts (You don't need to define the whole structure that is common to both several times).")
-    tag_ids = fields.Many2many('account.account.tag', 'account_account_template_account_tag', string='Account tag', help="Optional tags you may want to assign for custom reporting")
-
-    @api.depends('name', 'code')
-    def name_get(self):
-        res = []
-        for record in self:
-            name = record.name
-            if record.code:
-                name = record.code + ' ' + name
-            res.append((record.id, name))
-        return res
+    # --------------------------------------------------------------------------------
+    # Template selection
+    # --------------------------------------------------------------------------------
+
+    @lru_cache
+    def get_chart_template_mapping(self):
+        def templ(code, name=None, country='', modules=None, parent=None):
+            country_code = country or code.split('_')[0] if country is not None else None
+            country = country_code and self.ref(f"base.{country_code}", raise_if_not_found=False)
+            country_name = "%s %s" % ("".join(chr(int(f"1f1{ord(c)+165:02x}", base=16)) for c in country.code), f" {country.name}") if country else ''
```

Comment:
I beg your pardon?

Reply:
I am very proud of this code. It was actually written on the 24th of December on one of my off days.